fhir-remote-index/src/main/java/com/ibm/fhir/remote/index/app/Main.java

```diff
@@ -0,0 +1,442 @@
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.api.SchemaType;
+import com.ibm.fhir.database.utils.citus.CitusTranslator;
+import com.ibm.fhir.database.utils.citus.ConfigureConnectionDAO;
+import com.ibm.fhir.database.utils.common.JdbcConnectionProvider;
+import com.ibm.fhir.database.utils.model.DbType;
+import com.ibm.fhir.database.utils.postgres.PostgresPropertyAdapter;
+import com.ibm.fhir.database.utils.postgres.PostgresTranslator;
+import com.ibm.fhir.database.utils.thread.ThreadHandler;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.remote.index.api.IMessageHandler;
+import com.ibm.fhir.remote.index.cache.IdentityCacheImpl;
+import com.ibm.fhir.remote.index.database.CacheLoader;
+import com.ibm.fhir.remote.index.database.DistributedPostgresMessageHandler;
+import com.ibm.fhir.remote.index.database.PlainDerbyMessageHandler;
+import com.ibm.fhir.remote.index.database.PlainPostgresMessageHandler;
+import com.ibm.fhir.remote.index.kafka.RemoteIndexConsumer;
+import com.ibm.fhir.remote.index.sharded.ShardedPostgresMessageHandler;
+
+/**
+ * Main class for the FHIR remote index service Kafka consumer
+ */
+public class Main {
+    private static final Logger logger = Logger.getLogger(Main.class.getName());
+    // Properties holding the Kafka connection information
+    private final Properties kafkaProperties = new Properties();
+    // Properties holding the JDBC connection information
+    private final Properties databaseProperties = new Properties();
+
+    private String topicName;
+
+    // The standard consumer group which all the remote index consumers should use
+    private String consumerGroup = "remote-index-service-cg";
+
+    private int consumerCount = 1;
+
+    private Duration pollDuration = Duration.ofSeconds(10);
+    private long maxBatchCollectTimeMs = 5000;
+
+    // The max time we wait for the database to catch up with what was sent to Kafka
+    // Must be a little longer than the the Liberty transaction timeout
+    private long maxReadyTimeMs = 180000;
+
+    // the list of consumers
+    private final List<RemoteIndexConsumer> consumers = new ArrayList<>();
+
+    // track the number of consumers that are still running
+    private AtomicInteger stillRunningCounter;
+
+    private volatile boolean running = true;
+
+    // Exit if we drop below this number of running consumers
+    private float minRunningConsumerRatio = 0.5f;
+    private int minRunningConsumerThreshold;
+    private IdentityCacheImpl identityCache;
+
+    // Database Configuration
+    private SchemaType schemaType = SchemaType.PLAIN;
+    private IDatabaseTranslator translator;
+    private IConnectionProvider connectionProvider;
+    private DbType dbType = DbType.POSTGRESQL;
+    
+    /**
+     * Parse the given command line arguments
+     * @param args
+     */
+    public void parseArgs(String[] args) {
```

Comment:
Would like to see a README with info on how to use this.  Either in this PR or as part of #3697

Reply:
done